components/global/test_helpers.tsx

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {ProductComponent} from 'types/store/plugins';
+
+type TestProductComponent = Omit<ProductComponent, 'switcherIcon'> & {
+    switcherIcon: string;
+}
+
+export function makeProduct(name: string): TestProductComponent {
```

Comment:
If you think this might be used elsewhere, this could be put in `utils/test_helper` instead. We could wait until we actually need to use this elsewhere though.